packages/providers/upload-aws-s3/src/index.ts

```diff
@@ -23,146 +36,133 @@ interface File {
+  // Keep this for V5
+  credentials?: AwsCredentialIdentity;
+  params?: AWSParams;
+  [k: string]: any;
 }
 
-interface InitOptions extends Partial<AWS.S3.ClientConfiguration> {
+export type InitOptions = (DefaultOptions | { s3Options: DefaultOptions }) & {
   baseUrl?: string;
   rootPath?: string;
-  s3Options: AWS.S3.ClientConfiguration & {
-    params: {
-      Bucket: string; // making it required
-      ACL?: string;
-      signedUrlExpires?: string;
-    };
-  };
-}
+  [k: string]: any;
+};
 
-export default {
-  init({ baseUrl, rootPath, s3Options, ...legacyS3Options }: InitOptions) {
-    if (Object.keys(legacyS3Options).length > 0) {
-      process.emitWarning(
-        "S3 configuration options passed at root level of the plugin's providerOptions is deprecated and will be removed in a future release. Please wrap them inside the 's3Options:{}' property."
-      );
-    }
+const assertUrlProtocol = (url: string) => {
+  // Regex to test protocol like "http://", "https://"
+  return /^\w*:\/\//.test(url);
+};
+
+const getConfig = ({ baseUrl, rootPath, s3Options, ...legacyS3Options }: InitOptions) => {
+  if (Object.keys(legacyS3Options).length > 0) {
+    process.emitWarning(
+      "S3 configuration options passed at root level of the plugin's providerOptions is deprecated and will be removed in a future release. Please wrap them inside the 's3Options:{}' property."
+    );
+  }
+
+  const config = {
+    ...s3Options,
+    ...legacyS3Options,
+    credentials: extractCredentials({ s3Options, ...legacyS3Options }),
+  };
 
-    const config = { ...s3Options, ...legacyS3Options };
+  config.params.ACL = getOr(ObjectCannedACL.public_read, ['params', 'ACL'], config);
 
-    const S3 = new AWS.S3({
-      apiVersion: '2006-03-01',
-      ...config,
-    });
+  return config;
+};
 
+export default {
+  init({ baseUrl, rootPath, s3Options, ...legacyS3Options }: InitOptions) {
+    // TODO V5 change config structure to avoid having to do this
+    const config = getConfig({ baseUrl, rootPath, s3Options, ...legacyS3Options });
+    const s3Client = new S3Client(config);
     const filePrefix = rootPath ? `${rootPath.replace(/\/+$/, '')}/` : '';
 
     const getFileKey = (file: File) => {
       const path = file.path ? `${file.path}/` : '';
-
       return `${filePrefix}${path}${file.hash}${file.ext}`;
     };
 
-    const ACL = getOr('public-read', ['params', 'ACL'], config);
-
-    const upload = (file: File, customParams = {}): Promise<void> =>
-      new Promise((resolve, reject) => {
-        const fileKey = getFileKey(file);
-
-        if (!file.stream && !file.buffer) {
-          reject(new Error('Missing file stream or buffer'));
-          return;
-        }
-
-        const params = {
-          Key: fileKey,
+    const upload = async (file: File, customParams: Partial<PutObjectCommandInput> = {}) => {
+      const fileKey = getFileKey(file);
+      const parallelUploads = new Upload({
```

Comment:
```suggestion
      const parallelUploads = new Upload({
```
Can we rename this since it's just one file?